### ISSUE TITLE
Game crash on collision playing spooked world

### ISSUE THREAD

**Issue by sourcemonk** [ErikaRedmark/monkey-shines-java-port#78]
While playing spooks, the game crashed when Bonzo collided with a skull.

![bug_report_monkey_shines](https://cloud.githubusercontent.com/assets/4350563/8639736/9fcb9b20-28b0-11e5-8fca-173fba493454.jpeg)

Here's the stack trace:

 TRex in /media/meddling_monk/Carnivore/Downloads/monkeyShines
○ → java -jar lib/monkey_shines.jar 
Binary Location: /media/meddling_monk/Carnivore/Downloads/monkeyShines/lib/monkey_shines.jar
WARNING: Could not open configuration file. Dropping to defaults.
Reason: /media/meddling_monk/Carnivore/Downloads/monkeyShines/lib/monkey_shines.jar/logging.properties: Not a directory
File handler set up. Output is to console and 'monkeyshines.log'
Log level for system is INFO
Jul 12, 2015 4:05:33 PM org.erikaredmark.monkeyshines.logging.MonkeyShinesLog initialise
INFO: Monkey Shines Java Port Log File Initialised
Calculation of 60 got a gain of -11.869794032222648
Decibel offset for music: -21.869793
Calculation of 100 got a gain of -10.081904349041679
Decibel offset for sound: -10.081904
Exception in thread "AWT-EventQueue-0" java.lang.IllegalStateException: Can't get 40x40 draw location during a death animation
        at org.erikaredmark.monkeyshines.Bonzo.getDrawLocationInSprite(Bonzo.java:1140)
        at org.erikaredmark.monkeyshines.Sprite.pixelCollision(Sprite.java:534)
        at org.erikaredmark.monkeyshines.World.checkCollisions(World.java:685)
        at org.erikaredmark.monkeyshines.Bonzo.update(Bonzo.java:971)
        at org.erikaredmark.monkeyshines.GameWorldLogic$5.actionPerformed(GameWorldLogic.java:178)
        at javax.swing.Timer.fireActionPerformed(Timer.java:313)
        at javax.swing.Timer$DoPostEvent.run(Timer.java:245)
        at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
        at java.awt.EventQueue.access$500(EventQueue.java:97)
        at java.awt.EventQueue$3.run(EventQueue.java:709)
        at java.awt.EventQueue$3.run(EventQueue.java:703)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
        at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
        at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
        at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

My Java version is: 
 TRex in /media/meddling_monk/Carnivore/Downloads/monkeyShines
○ → java -version
java version "1.8.0_45"
Java(TM) SE Runtime Environment (build 1.8.0_45-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.45-b02, mixed mode)

I'm running Kubuntu 64-bit:
 TRex in /media/meddling_monk/Carnivore/Downloads/monkeyShines
○ → uname -a
Linux TRex 3.19.0-22-generic #22-Ubuntu SMP Tue Jun 16 17:15:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

**Comment by ErikaRedmark** [comment]
The issue was caused by a failure to break out of further collision detection after bonzo is already dead (which is triggered here due to being within the bounding boxes of two sprites at once, despite the fact that the second was not a complete pixel-collision). I've replicated this issue and fixed it in the latest master. I will hold off on making a 1.1 release whilst I do more playtesting, and also give you more time to find any more bugs. Thank you.
